client/dashboard/sites/performance/performance-insight-table.tsx

```diff
@@ -0,0 +1,150 @@
+import { __, sprintf } from '@wordpress/i18n';
+import Markdown from 'react-markdown';
+import { Text } from '../../components/text';
+import { getFormattedNumber, getFormattedSize } from '../../utils/site-performance';
+import { InsightScreenshotWithOverlay } from './performance-insight-screenshot';
+import type {
+	PerformanceMetricsDetailsQueryResponse,
+	PerformanceMetricsDetailsItem,
+} from './types';
+import type { SitePerformanceReport } from '@automattic/api-core';
+
+const renderNode = (
+	data: { [ key: string ]: any },
+	fullPageScreenshot: SitePerformanceReport[ 'fullPageScreenshot' ]
+) => {
+	const rect = data.boundingRect;
+
+	if ( fullPageScreenshot && rect && rect.width !== 0 && rect.height !== 0 ) {
+		const maxThumbnailSize = { width: 147, height: 100 };
+		return (
+			<InsightScreenshotWithOverlay
+				nodeId={ data.lhId }
+				screenshot={ fullPageScreenshot.screenshot }
+				elementRectSC={ rect }
+				maxRenderSizeDC={ maxThumbnailSize }
+			/>
+		);
+	}
+	return (
+		<div>
+			<p>{ data?.nodeLabel }</p>
+			<code>{ data?.snippet }</code>
+		</div>
+	);
+};
+
+const PerformanceInsightTable = ( {
+	details,
+	fullPageScreenshot,
+}: {
+	details: PerformanceMetricsDetailsQueryResponse;
+	fullPageScreenshot: SitePerformanceReport[ 'fullPageScreenshot' ];
+} ) => {
+	const { headings = [], items = [] } = details ?? {};
+	const fields = headings.map( ( heading ) => ( {
+		id: heading.key,
+		label: heading.label,
+		enableSorting: false,
+		enableHiding: false,
+		render: ( { item }: { item: PerformanceMetricsDetailsItem } ) => {
+			const value =
+				heading.subItemsHeading && item.__isSubItem
+					? item[ heading.subItemsHeading.key ]
+					: item[ heading.key ];
+
+			const valueType =
+				heading.subItemsHeading && item.__isSubItem
+					? heading.subItemsHeading.valueType ?? heading.valueType
+					: heading.valueType;
+
+			if ( typeof value === 'object' ) {
+				switch ( value?.type ) {
+					case 'node':
+						return renderNode( value, fullPageScreenshot );
+
+					case 'code':
+						return (
+							<div>
+								<pre>
+									<code>{ value?.value }</code>
+								</pre>
+							</div>
+						);
+					case 'numeric':
+						return getFormattedNumber( value.value );
+					case 'url':
+					case 'source-location':
+						if ( typeof value.location === 'object' ) {
+							return [ value.location.url, value.location.line, value.location.column ].join( ':' );
+						}
+						return value?.url;
+				}
+
+				return value?.value;
+			}
+
+			if ( typeof value === 'string' || typeof value === 'number' ) {
+				switch ( valueType ) {
+					case 'ms':
+					case 'timespanMs':
+						// translators: %(ms)d is the value to be displayed in milliseconds.
+						return sprintf( __( '%(ms)dms' ), { ms: getFormattedNumber( value ) } );
+					case 'bytes':
+						return getFormattedSize( Number( value ) || 0 );
+					case 'numeric':
+						return getFormattedNumber( value, 2 );
```

Comment:
2 is the default value so it should be good to omit.
```suggestion
						return getFormattedNumber( value );
```